src/main/java/graphql/util/TreeTransformerUtil.java

```diff
@@ -20,17 +21,28 @@
      */
     public static <T> TraversalControl changeNode(TraverserContext<T> context, T changedNode) {
         NodeZipper<T> zipperWithChangedNode = context.getVar(NodeZipper.class).withNewNode(changedNode);
-        List<NodeZipper<T>> zippers = context.getSharedContextData();
         boolean changed = context.isChanged();
-        if (changed) {
-            // this is potentially expensive
-            replaceZipperForNode(zippers, context.thisNode(), changedNode);
-            context.changeNode(changedNode);
+        if (context.isParallel()) {
+            Queue<NodeZipper<T>> zippers = context.getSharedContextData();
+            if (changed) {
+                throw new RuntimeException("multiple changes per node not supported for parallel");
```

Comment:
consistency of error message from above

`multiple changes per node is not supported for parallel traversing`